p2p/pex/known_address.go

```diff
@@ -54,6 +55,14 @@ func (ka *knownAddress) markGood() {
 	ka.LastSuccess = now
 }
 
+func (ka *knownAddress) ban() {
+	ka.LastBanTime = time.Now()
+}
+
+func (ka *knownAddress) isBanned(banTime time.Duration) bool {
```

Comment:
I don't think it's necessary to pass banTime as a parameter here, maybe just use the constant directly.

Reply:
I was considering whether we want to make it something that is either a variant - or at least configurable

Reply:
Yeah, that makes sense. But I don't think it's something that all callers should have to deal with every time time they want to check if someone is banned, it's more convenient if `isBanned()` just takes all conditions into account automatically. But I'm not sure what the best way is to inject variables into the current architecture - I guess this is fine for a first iteration.

Reply:
One option may be to have `ban()` take a parameter for how long to ban, and store `bannedUntil` as the time when the ban ends (the max of the current and given). Then `isBanned()` can simply compare `time.Now()` against `bannedUntil`.

Reply:
yes that could be a good idea

Reply:
Do we want to have increasing the ban time i.e. keep count of the amount of bans and therefore increase the duration based on the amount of bans? or is that unnecessary complexity

Reply:
I don't think that's necessary in the first iteration. But it did occur to me that passing the ban time to `ban()` would allow us to vary the ban duration for different offences, if that's something we would need later.

Reply:
Would we perhaps want to expose `Ban()` so other modules could use it?

Reply:
Not sure. Let's consider than when there's an actual need for it.

Reply:
Okay we can leave it for later. I was talking to Tess how each module sort of has their own way of categorising and dealing with peers that makes things at a whole inconsistent and potentially repetitive

Reply:
For sure. That's something to consider for the p2p refactor.